torch/linalg/__init__.py

```diff
@@ -633,38 +679,39 @@
 The drivers `'gelsy'`, `'gelsd'`, `'gelss'` handle rank-deficient inputs.
 See the note below on how to choose the best driver. See also the `full description of these drivers`_
 
-:attr:`cond` is used to determine the effective rank of the matrices in :attr:`input`
+:attr:`cond` is used to determine the effective rank of the matrices in :attr:`A`
 for rank-revealing drivers, that is, when :attr:`driver` is one of (`'gelsy'`, `'gelsd'`, `'gelss'`).
 In this case, if :math:`\sigma_i` are the singular values of `A` in decreasing order,
 :math:`\sigma_i` will be rounded down to zero if :math:`\sigma_i \leq \text{cond} \cdot \sigma_1`.
-If :attr:`cond` `= None` (default), :attr:`cond` is set to the machine precision of the dtype of :attr:`input`.
+If :attr:`cond` `= None` (default), :attr:`cond` is set to the machine precision of the dtype of :attr:`A`.
 
-This function returns the solution to the problem and some extra information in a namedtuple of
+This function returns the solution to the problem and some extra information in a named tuple of
 four tensors `(solution, residuals, rank, singular_values)` containing:
 
 - `solution`: the least squares solution
 - `residuals`: the squared residuals of the solutions, that is :math:`\|AX - B\|_F^2`.
   It is computed when `m > n` and the matrix is full-rank,
   otherwise, it is an empty tensor.
-- `rank`: tensor of ranks of the matrices in :attr:`input`
-  It has shape equal to the batch dimensions of :attr:`input`.
+- `rank`: tensor of ranks of the matrices in :attr:`A`
+  It has shape equal to the batch dimensions of :attr:`A`.
   It is computed when :attr:`driver` is one of (`'gelsy'`, `'gelsd'`, `'gelss'`),
   otherwise it is an empty tensor.
-- `singular_values`: tensor of singular values of the matrices in :attr:`input`.
+- `singular_values`: tensor of singular values of the matrices in :attr:`A`.
   It has shape `(*, min(m, n))`.
   It is computed when :attr:`driver` is one of (`'gelsd'`, `'gelss'`),
   otherwise it is an empty tensor.
 
 .. note::
-    The solution to this problem can be computed in terms of the Moore-Penrose pseudoinverse of
-    `A` as `A.pinv() @ B`.
-    For this reason, this function is a faster and more stable way of computing `A.pinv() @ B`.
+    A solution to this problem can be computed in terms of the pseudoinverse
```

Comment:
This note reads a little oddly to me. The idea is that lstsq should be preferred to using the pseudo inverse, right? In that case I'd try:

"While solution = a.pinv() @ b, this function computes the solution in a faster and more numerically stable way that performing the computations separately."

Reply:
Corrected here and in `solve`.